modules/apps/29-fee/ibc_module_test.go

```diff
@@ -527,11 +527,18 @@ func (suite *FeeTestSuite) TestOnRecvPacket() {
 
 			case tc.forwardRelayer && result == nil:
 				suite.Require().Equal(nil, result)
+				packetId := channeltypes.NewPacketId(packet.GetDestChannel(), packet.GetDestPort(), packet.GetSequence())
+
+				// retrieve the forward relayer that was stored in `onRecvPacket`
+				relayer, _ := suite.chainB.GetSimApp().IBCFeeKeeper.GetRelayerAddressForAsyncAck(suite.chainB.GetContext(), packetId)
+				suite.Require().Equal(relayer, suite.chainA.SenderAccount.GetAddress().String())
 
 			case !tc.forwardRelayer:
+				forwardAddr, _ := suite.chainB.GetSimApp().IBCFeeKeeper.GetCounterpartyAddress(suite.chainB.GetContext(), suite.chainA.SenderAccount.GetAddress().String(), suite.path.EndpointA.ChannelID)
+
 				expectedAck := types.IncentivizedAcknowledgement{
 					Result:                ibcmock.MockAcknowledgement.Acknowledgement(),
-					ForwardRelayerAddress: "",
+					ForwardRelayerAddress: forwardAddr,
```

Comment:
this confuses me a little, shouldn't this case have an empty forward relayer?

Reply:
Yes it should you're right. I need to make a change to our calls to get/set CounterpartyAddress

Reply:
I've updated this now so that anywhere we change DestChannel/Port to Source we'll get failing tests (for forward relayer or counterparty)